### DISCUSSION TITLE
Why is portable not updating to 1.23.1 when running the update.bat (win11)?

### DISCUSSION THREAD

**Discussion by stephantual** [Comfy-Org/ComfyUI-Manager#298] [Q&A] [answered]
Trying to understand the release process here. I'm using portable because It's very convenient to store the whole folder in a 'store only' 7zip and pass it between machines as i go on the road. 

Problem: I downloaded 1.19.1 and i know the latest is 1.23.1 as of today january 5th 2024. Running 'update.bat' returns that 'comfy is up to date', when it clearly isn't.

I would hate to have to restart from scratch and git clone it and reinstall everything, manager or not (because I use a lot of custom stuff and made a lot of modifications to my folders, including attempting to create files to 'fix' certain broken 3rd party nodes.

TLDR: how do i upgrade if  use protable and if the answer is update.bat, why doens't it work?

**Comment by ltdrdata** [comment]
In the very early stages, the portable version had incorrect permissions included during packaging, resulting in an update issue due to permission problems. If you are using that version, it is recommended to reinstall with the new portable version.

If you don't want to reinstall, you'll need to modify the files related to config and credentials inside the .git directory.

The most reliable method is to override the `.git` directory with the one from the freshly cloned ComfyUI, perform `git reset --hard`, remove any remaining untracked files to ensure a conflict-free state, and then execute `git pull`.

And this is an issue unrelated to ComfyUI-Manager.

**Comment by stephantual** [comment]
1.19 was downloaded 3 days before xmas :) I love how fast this moves - you're an absolute superstar 💯 . And thank you for confirming i'm not crazy. One comment (which you may have no control over) - I got flamed on the reddit when I suggested this could be an issue, and it was implied i was an 'idiot' for wanting to reinstall because 'update.bat duuuuh' . This is my opportunity to humbly (if you have any say in the matter) to have the reddit upgraded with flair, proper moderation, and some stickies - it's a mess in there and I do most of my communications on discord or the element channel at this point, where people are far more kind and considerate.

**Reply by ltdrdata** [reply]
The issue that occurred in the very early versions of the portable version unfortunately continues to persist to this day. Sometimes, things that were mistakenly distributed can become a persistent headache.

**Reply by stephantual** [reply]
Thank you @ltdrdata . So that this helps others in a similar situation, is it fair to say that my copy of confyui is still deemed to be 'the portable one'? The reason is that i transfer the entire folder back and forth between two machines when i'm on the go, which is very often.

I have followed the steps: 
- cloned the repo in a separate folder
- deleted the old .git from the previous install
- replaced it with *just* .git from the new cloned repo
- ran git reset --hard
- there were no conflics that I could tell
- git pulled
... now it starts with '### ComfyUI Revision: 1881 [af94eb14] | Released on '2024-01-06' < which I hope is the correct one.

It was extremely fast, almost instant in fact, so I'm wondering if i have the 'right' files in there. Since i don't have a reference point to look at or two files to compare (beause i don't know what changed), i can only hope I did the right thing :)

let me know your thoughts, because I can't be the only in this situation.
Bonus question: if i was to installl comfy in 'non portble' mode, given that python seems embedded either way, why is there a difference? Thank you 🙏

**Reply by ltdrdata** [reply, marked as answer]
You can check validity by `git status`.
If there are no modifications, it is normal.

Files like `comfyui.log` that do not belong to ComfyUI may appear as untracked files, but it should not be a problem as long as they do not cause conflicts when performing `git pull`.

**Reply by stephantual** [reply]
Clean tree, happy days, you're a star!
